horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/RunServiceImpl.java

```diff
@@ -1004,4 +1012,213 @@ public DatasetList listDatasetsBySchema(String uri, Integer limit, Integer page,
       DatasetList list = new DatasetList();
       return list;
    }
+
+   private void transform(Run run, String role, String token) {
+      if (run == null || role == null) {
+         log.errorf("Transformation parameters error: run %s, role %s", run, role);;
+      }
+
+      Collection<RunService.RunSchema> runSchemas = getSchemas(run, role, token);
+
+      try (@SuppressWarnings("unused") CloseMe h1 = roleManager.withRoles(em, role);
+            @SuppressWarnings("unused") CloseMe h2 = roleManager.withToken(em, token)) {
+
+         Test test = Test.findById(run.testid);
+         boolean matched = false;
+         List<JsonNode> transformedSingleNodeResults = new ArrayList<>();
+         JsonNode transformedArrayNodeResult = null;
+         List<DataSet> dataSets = new ArrayList<>();
+         int ordinal = 0;
+         boolean functionError = false;
+         if (!runSchemas.isEmpty()){
+
+            for ( RunService.RunSchema s : runSchemas ){
+
+               Collection<Transformer> transformers = test.transformers;
+               if (transformers.isEmpty()) {
+                  dataSets.add(new DataSet(run.start, run.stop, "",
```

Comment:
If the test has no transformers, the result should be **one dataset** with all schemas, not an individual dataset for each schema.

Reply:
ok